projects/transformers/run.py

```diff
@@ -441,16 +471,28 @@ def model_init():
     # Update any extra kwargs defined in config
     hp_search_kwargs.update(**model_args.hp_extra_kwargs)
 
-    # Run hp search and save results
-    best_run = trainer.hyperparameter_search(**hp_search_kwargs)
-    logging.info(f"Best run: {best_run}")
+    # Run hp search and save results. Code to remove checkpoints won't get
+    # called if ANY of the trials error out, so wrap with try/except.
+
+    success = False
+    try:
+        best_run = trainer.hyperparameter_search(**hp_search_kwargs)
+        logging.info(f"Best run: {best_run}")
+        success = True
+    except TuneError:
+        logging.info(f"One or more trials errored out")
+    finally:
+        logging.info("An unknown error occured during hp search.")
 
     # Delete all saved models and checkpoints to save space. All we need
     # are the params and scores. Currently this is a hack that is specific
     # to ray. May need to modify so that it serves the same function for
     # sigopt, once I integrate it.
     rm_prefixed_subdirs(training_args.output_dir, "run-")
 
+    if not success:
+        return {}
```

Comment:
Return is {} anyway - this might be better arranged with a `if success: code` and a single return at the end. The return here would imply if not success you would have a different output. 

Also, it is a bit weird that this function always returns an empty dict. Shouldn't it be returning the eval results instead?

Reply:
It is a little funky. It returns {} because run_finetuning_multiple_tasks sets up aloop over tasks / runs, and calls run_finetuning_single_task or run_finetuning_single_task_with_hp_search inside it. The former returns task results. The latter is just used to find good hyperparams and is not intended (currently) for producing finetuning results, so it returns something empty that is the same type. The reason for returning early is that if hp search didn't complete, best_run won't be defined and the rest of the code will break.

This raises two changes: 1) I can just define best_run with a null value so the "if not success" statement can be removed, or simply delete the code to save best_run to a text file since this is used for redundancy, and is not the primary way of analyzing hp search results. 2) The thing returned in run_finetuning_single_task, eval_results, isn't even defined unless training_args.do_evaluate. I haven't seen the code break because of this, since do_eval is the default and there are checks / warnings elsewhere, but it needs fixed.